docs/slicer_cli_plugins.rst

```diff
@@ -0,0 +1,255 @@
+
+Value Attributes
+~~~~~~~~~~~~~~~~
+
+Some value nodes can have additional attributes.  These are:
+
+* ``multiple``: is a binary value of ``true`` or ``false``.  A value where multiple is true can be specified multiple times.
+
+* ``coordinateSystem``: this is used for ``point`` and ``pointfile`` values and is ignored by slicer_cl_web.
+
+* ``fileExtensions``: a comma-separated list of expected file extensions for any of the ``file``, ``image``, or ``item`` values.
+
+* ``reference``: this is mostly a slicer_cli_web extension.  For value types of ``directory``, ``file``, ``image``, or ``item``, if this is ``_girder_id_``, then value will be passed as a Girder ID string rather than converted to a Girder resource.  Otherwise, if this is the ``name`` of another value, the current parameter value will reference the other value.  This can be used, for instance, to associate annotations with parent images.
+
+* ``subtype``: used by ``image`` and ``geometry`` values.  slicer_cli_web ignores this value.
+
+* ``shapes``: used by ``region`` values; a slicer_cli_web extension.  A comma-separated list of values that can include ``default``, ``rectangle``, ``polygon``, ``line``, ``polyline``, and ``point``, plus ``multi`` and one of ``submit`` (or ``submitoff``), ``submiton``, or ``autosubmit``. In the official schema, region is a vector of six values of the form x,y,z,rx,ry,rz, defining a rectangle based on its center and radius in each of three dimensions.  This is the ``default`` shape.  The ``rectangle`` shape allows a vector of four values defining a rectangle of the form x,y,width,height, where x,y is the left and top of the rectangle in pixel coordinates.  Many algorithms that accept this value accept -1,-1,-1,-1 as a default to specify the whole conceptual space.  The ``polygon`` shape allows for a list of x,y values.  Polygons must always have at least four points so that the vector of values cannot be confused with the default; repeat the first vertex at the end to specify a triangle.  The ``line`` shape allows a two-vertex line.  To disambiguate this from a rectangle, the values -2,-2 are added after the line.  The ``polyline`` shape allows a multi vertex line, indicated again by a -2,-2 value after the line.  A ``point`` is a single vertex.  ``multi`` allow multiple shapes, indicated by separating coordinates of each shape by -1,-1.  Note that neither -1,-1 nor -2,-2 are allowed as coordinates within a shape -- to use those, specify them with decimals (e.g., -1.0,-1.0).  The submit options will add suggestions on how the UI should handle changes.  If present, the option to auto-run a job as soon as a valid shape is set should be present.  ``autosubmit`` means this should always happen.  ``submit`` or ``submitoff`` offers this as a setting but is default to not submit the job.  ``submiton`` offers this as a setting and defaults to submitting the job.
```

Comment:
I think splitting this and other attributes into a nested list would make this section easier to read.